src/main/java/backend/Logic.java

```diff
@@ -68,22 +68,26 @@ public Logic(UIManager uiManager, Preferences prefs, boolean isTestMode, boolean
         return repoIO.isRepositoryValid(repoId);
     }
 
-    public void refresh() {
-        String message = "Refreshing " + models.toModels().stream()
-            .map(Model::getRepoId)
-            .collect(Collectors.joining(", "));
-
-        logger.info(message);
-        UI.status.displayMessage(message);
-
-        Futures.sequence(models.toModels().stream()
-            .map(repoIO::updateModel)
-            .collect(Collectors.toList()))
-                .thenApply(models::replace)
-                .thenRun(this::updateUI)
-                .thenCompose(n -> getRateLimitResetTime())
-                .thenApply(this::updateRemainingRate)
-                .exceptionally(Futures::log);
+    public void refresh(boolean isNotificationPaneShowing) {
+        if (!isNotificationPaneShowing) {
```

Comment:
We could add a TODO comment here, so we know to revert it in future. Also the simpler branch can come first with a guard clause.
```java
if (isNotificationPaneShowing) {
    // log
    return;
}
// rest
```